### DISCUSSION TITLE
Multiple windows with different views of the same Neovim instance?

### DISCUSSION THREAD

**Discussion by smackesey** [akiyosi/goneovim#546] [Q&A] [answered]
I asked this question in the neovim repo discussions: https://github.com/neovim/neovim/discussions/29994

>0.9.x brought the --remote-ui option, which allows for opening two OS terminal windows against the same Neovim instance. But is it possible to have those OS windows have different “views” (i.e. different collections of vim tabpages/windows)?
>
>My primary motivation is to run a debugger and show the UI in a separate OS window/display.

A neovim dev gave me an answer (emphasis mine):

>This is not achievable using --remote-ui. **This is only achievable with an ext_multigrid GUI (e.g. goneovim)** using external floating windows (using methods mentioned in :h nvim_open_win() or :h CTRL-W_ge).

So is this currently possible in Goneovim? If so, are there instructions somewhere for doing it. If not, do we expect it to be available any time soon?

**Comment by akiyosi** [comment]
@smackesey 
Hi :)
The goneovim has the ability to create an external float window. This can be done in two main approaches:

1. create it with nvim_open_win
e.g.
:lua vim.api.nvim_open_win(vim.api.nvim_create_buf(false, true), false, { external=true, width=50, height=10})

2. making an existing window into an external window
This can be accomplished with Ctrl-w + ge

**Reply by smackesey** [reply]
Wow, it works, very cool! But I played around a little and immediately encountered the limitations that:

- The external window doesn't seem to be able to hold splits (if I run `vsplit` it splits a window in the "master" OS window
- The external window doesn't seem to be able to hold tabpages (if I run `tabedit` it adds a tab in the "master" OS window)

Any way around these limitations? I looked at the ui-multigrid docs and didn't see anything relevant.

It's cool that you can make a single vim window into another OS window, but that's not really what I'm after here-- I 'd like two "full" Vim UI windows working off the same set of open buffers.

**Comment by akiyosi** [comment, marked as answer]
Thank you for your feedback. The limitations you encountered with the external window not holding splits or tabpages are indeed due to the current constraints of Neovim's UI features. To have two full Vim UI windows working off the same set of open buffers, with each able to manage splits and tabpages independently, it would require modifications to the Neovim core. The current implementation of the ui-multigrid feature does not support this level of window management, and achieving this functionality would likely necessitate a significant overhaul of how Neovim handles UI interactions.

**Reply by smackesey** [reply]
Ok, thank you for your prompt answer-- still a cool feature.
